packages/components/src/components/post-pagination/post-pagination.tsx

```diff
@@ -0,0 +1,844 @@
+    this.validateProp('labelPage', 'string', true);
+  }
+  
+  @Watch('labelFirst')
+  validateLabelFirst() {
+    this.validateProp('labelFirst', 'string', true);
+  }
+  
+  @Watch('labelLast')
+  validateLabelLast() {
+    this.validateProp('labelLast', 'string', true);
+  }
+  
+  @Watch('disabled')
+  validateDisabled() {
+    this.validateProp('disabled', 'boolean', false);
+  }
+  
+  @Watch('page')
+  @Watch('pageSize')
+  @Watch('collectionSize')
+  handlePropsChange() {
+    this.updatePagesWithValidation();
+  }
+
+  componentWillLoad() {
+    this.paginationId = `pagination-${this.host.id || nanoid(6)}`;
+
+    if (this.page == null) {
+      this.page = 1;
+    }
+
+    this.runAllValidations();
+  }
+
+  componentDidLoad() {
+    this.connected = true;
+    
+    this.debouncedResize = debounce(this.handleResizeInternal.bind(this), RESIZE_DEBOUNCE_MS);
+    
+    window.addEventListener('resize', this.debouncedResize);
+    this.lastWindowWidth = window.innerWidth;
+    
+    this.scheduleMeasurement();
+  }
+
+  disconnectedCallback() {
+    this.connected = false;
+    
+    if (this.debouncedResize) {
+      window.removeEventListener('resize', this.debouncedResize);
+    }
+    
+    if (this.measurementTimeoutId !== null) {
+      clearTimeout(this.measurementTimeoutId);
+      this.measurementTimeoutId = null;
+    }
+  }
+
+  /**
+   * Validate a prop with the appropriate check function
+   */
+  private validateProp(propName: ValidatableProp, type: 'string' | 'number' | 'boolean', required: boolean = true) {
+    if (required) {
+      checkRequiredAndType(this, propName, type);
+    } else {
+      checkEmptyOrType(this, propName, type);
+    }
+  }
+
+  /**
+   * Run all prop validations
+   */
+  private runAllValidations() {
+    this.validateProp('page', 'number', false);
+    this.validateProp('pageSize', 'number', true);
+    this.validateProp('collectionSize', 'number', true);
+    this.validateProp('label', 'string', true);
+    this.validateProp('labelPrevious', 'string', true);
+    this.validateProp('labelNext', 'string', true);
+    this.validateProp('labelPage', 'string', true);
+    this.validateProp('labelFirst', 'string', true);
+    this.validateProp('labelLast', 'string', true);
+    this.validateProp('disabled', 'boolean', false);
+  }
+
+  /**
+   * Schedule measurement attempt with timeout
+   */
+  private scheduleMeasurement() {
+    if (!this.connected) {
+      return;
+    }
+    
+    this.measurementTimeoutId = window.setTimeout(() => {
+      this.attemptMeasurement();
+    }, MEASUREMENT_DEBOUNCE_MS);
+  }
+
+  /**
+   * Attempt to measure, reschedule if not ready
+   */
+  private attemptMeasurement() {
+    if (!this.connected) {
+      return;
+    }
+    
+    const canMeasure = this.navRef?.clientWidth > 0 && this.hiddenItemsRef;
+    
+    if (canMeasure) {
+      this.measureAndCalculateVisiblePages();
+    } else {
+      this.scheduleMeasurement();
+    }
+  }
```

Comment:
Since you only call this method in the one above, you could nest it within the one above, which would automatically create a clearer scope and allow you to remove the redundant guard in the inner function.

```ts
private scheduleMeasurement() {
  if (!this.connected) {
    return;
  }
  
  this.measurementTimeoutId = window.setTimeout(attemptMeasurement, MEASUREMENT_DEBOUNCE_MS);
  
  // since this function lifes in the scope of its parent, you still have access to `this` within ;)
  function attemptMeasurement() {
    const canMeasure = ...
    ...
  }
}
```